wp-includes/sqlite-ast/class-wp-sqlite-driver.php

```diff
@@ -2592,6 +2887,215 @@ private function translate_show_like_or_where_condition( WP_Parser_Node $like_or
+	 *   4. NOT NULL + DEFAULT:    No value saves DEFAULT.
+	 *                             NULL is rejected on INSERT, but saves IMPLICIT DEFAULT on UPDATE.
+	 *                             DEFAULT saves DEFAULT.
+	 *
+	 * For more information about IMPLICIT DEFAULT values in MySQL, see:
+	 *   https://dev.mysql.com/doc/refman/8.4/en/data-type-defaults.html#data-type-defaults-implicit
+	 *
+	 * @param  string         $table_name The name of the target table.
+	 * @param  WP_Parser_Node $node       The "insertQueryExpression" or "insertValues" AST node.
+	 * @return string                     The translated INSERT query body.
+	 */
+	private function translate_insert_or_replace_body_in_non_strict_mode(
+		string $table_name,
+		WP_Parser_Node $node
+	): string {
+		// 1. Get column metadata from information schema.
+		$is_temporary  = $this->information_schema_builder->temporary_table_exists( $table_name );
+		$columns_table = $this->information_schema_builder->get_table_name( $is_temporary, 'columns' );
+		$columns       = $this->execute_sqlite_query(
+			"
+				SELECT column_name, is_nullable, column_default, data_type, extra
+				FROM $columns_table
+				WHERE table_schema = ?
+				AND table_name = ?
+				ORDER BY ordinal_position
+			",
+			array( $this->db_name, $table_name )
+		)->fetchAll( PDO::FETCH_ASSOC );
+
+		// 2. Get the list of fields explicitly defined in the INSERT statement.
+		$insert_list = array();
+		$fields_node = $node->get_first_child_node( 'fields' );
+		if ( $fields_node ) {
+			// This is the optional "INSERT INTO ... (field1, field2, ...)" list.
+			foreach ( $fields_node->get_child_nodes() as $field ) {
+				$insert_list[] = $this->unquote_sqlite_identifier( $this->translate( $field ) );
+			}
+		} else {
+			// When no explicit field list is provided, all columns are required.
+			foreach ( array_column( $columns, 'COLUMN_NAME' ) as $column_name ) {
+				$insert_list[] = $column_name;
+			}
+		}
+
+		// 3. Get the list of column names returned by VALUES or SELECT clause.
+		$select_list = array();
+		if ( 'insertQueryExpression' === $node->rule_name ) {
+			// When inserting from a SELECT query, we don't know the column names.
+			// Let's wrap the query with a SELECT (...) LIMIT 0 to get obtain them.
+			$expr = $node->get_first_child_node( 'queryExpressionOrParens' );
+			$stmt = $this->execute_sqlite_query(
+				'SELECT * FROM (' . $this->translate( $expr ) . ') LIMIT 1'
+			);
+			$stmt->execute();
+
+			for ( $i = 0; $i < $stmt->columnCount(); $i++ ) {
+				$select_list[] = $stmt->getColumnMeta( $i )['name'];
+			}
+		} else {
+			// When inserting from a VALUES list, SQLite uses "columnN" naming.
+			foreach ( array_keys( $insert_list ) as $position ) {
+				$select_list[] = 'column' . ( $position + 1 );
+			}
+		}
+
+		// 4. Compose a new INSERT field list with all columns from the table.
+		$fragment = '(';
+		foreach ( $columns as $i => $column ) {
+			$fragment .= $i > 0 ? ', ' : '';
```

Comment:
Nothing actionable for this PR but man, I wish we had a more convenient way of building these SQL queries – both for us here and for WordPress core. cc @dmsnell – we've discussed this recently